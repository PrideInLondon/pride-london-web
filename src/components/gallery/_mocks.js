export const mockData = [
  {
    name: 'Test image 1',
    originalAlt: 'foo',
    original: 'foo',
    description: 'Test decription',
    thumbnailAlt: 'foo',
    thumbnail: 'foo',
    photographer: {
      name: 'Test photographer 1',
      url: 'example.com',
    },
  },
]
