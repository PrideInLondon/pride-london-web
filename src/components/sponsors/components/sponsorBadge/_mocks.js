export const mockSponsor = {
  level: 'Headline',
  logo: '/logo1',
  name: 'Sponsor One',
  url: 'https://sponsor1.com',
}
