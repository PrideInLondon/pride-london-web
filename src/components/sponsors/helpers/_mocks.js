export const mockSponsors = [
  {
    level: 'Headline',
    logo: '/logo1',
    name: 'Sponsor One',
    url: 'https://sponsor1.com',
  },
  {
    level: 'Silver',
    logo: '/logo2',
    name: 'Sponsor Two',
    url: 'https://sponsor2.com',
  },
  {
    level: 'Silver',
    logo: '',
    name: 'Sponsor Three',
    url: 'https://sponsor3.com',
  },
]
