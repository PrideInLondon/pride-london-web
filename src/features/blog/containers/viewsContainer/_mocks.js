export const views = [
  {
    author: {
      display_name: {
        display_name: 'Test',
      },
    },
    category: 'Views',
    datePublished: '2019-05-17',
    headerImage: {
      file: {
        url: 'src',
      },
    },
    id: '1',
    title: 'test',
  },
  {
    author: {
      display_name: {
        display_name: 'Test',
      },
    },
    category: 'Views',
    datePublished: '2019-05-17',
    headerImage: {
      file: {
        url: 'src',
      },
    },
    id: '2',
    title: 'test',
  },
]
