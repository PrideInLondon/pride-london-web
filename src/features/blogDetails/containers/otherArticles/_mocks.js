export default {
  allContentfulArticle: {
    edges: [
      {
        node: {
          id: '73369e14-9fc7-5bec-a2ed-e3630d062d6b',
          title: 'A Featured Article',
          datePublished: '2019-08-14',
          category: 'Partners',
        },
      },
      {
        node: {
          id: '1a0304c1-9e37-5adc-9e7c-7a67adc8aeba',
          title: 'Hey hooo hola',
          datePublished: '2019-06-11',
          category: 'Partners',
        },
      },
      {
        node: {
          id: 'bf65a5ff-8b99-5d86-b0eb-1dabb58e0208',
          title: 'Yada yada yada',
          datePublished: '2019-05-17',
          category: 'Partners',
        },
      },
    ],
  },
}
