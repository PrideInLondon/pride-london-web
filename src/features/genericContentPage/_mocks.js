export const buttonNode = {
  data: {
    target: {
      fields: {
        buttonContent: { 'en-GB': 'Button primary type' },
        buttonHref: { 'en-GB': 'https://prideinlondon.org/who-we-are' },
        buttonPrimary: { 'en-GB': true },
      },
      sys: {
        contentType: {
          sys: {
            id: 'button',
          },
        },
      },
    },
  },
}

export const buttonNullContentNode = {
  data: {
    target: {
      fields: {
        buttonContent: undefined,
      },
      sys: {
        contentType: {
          sys: {
            id: 'button',
          },
        },
      },
    },
  },
}

export const videoNode = {
  data: {
    target: {
      fields: {
        title: { 'en-GB': 'Video for example' },
        videoId: { 'en-GB': '7ge36or-nz4' },
      },
      sys: {
        contentType: {
          sys: {
            id: 'video',
          },
        },
      },
    },
  },
}

export const imageNode = {
  data: {
    target: {
      fields: {
        title: { 'en-GB': 'Hello World' },
        description: {
          'en-GB': 'Image description',
        },
        file: {
          'en-GB': {
            contentType: 'image/jpeg',
            details: {
              image: { width: 5152, height: 3864 },
              size: 8068085,
            },
            fileName: 'image.jpg',
            url: 'src',
          },
        },
      },
    },
  },
}

export const typeformButtonNode = {
  data: {
    target: {
      fields: {
        buttonContent: { 'en-GB': 'Button typeform type' },
        buttonHref: { 'en-GB': 'https://prideinlondon.typeform.com/to/Z3WuCL' },
        buttonPrimary: { 'en-GB': true },
      },
      sys: {
        contentType: {
          sys: {
            id: 'button',
          },
        },
      },
    },
  },
}

export const notRecognizedNode = {
  data: {
    target: {
      fields: {},
      sys: {
        contentType: {
          sys: {
            id: 'not-recognized',
          },
        },
      },
    },
  },
}

export const sponsorSectionNode = {
  data: {
    target: {
      fields: {
        displayName: { 'en-GB': 'Test Sponsor Section' },
        sponsors: {
          'en-GB': [
            {
              fields: {
                sponsorName: {
                  'en-GB': 'Virtucon',
                },
                sponsorUrl: {
                  'en-GB': 'https://m.popkey.co/666ded/1VRvJ.gif',
                },
                sponsorLogo: {
                  'en-GB': {
                    fields: {
                      file: {
                        'en-GB': {
                          url:
                            '//images.ctfassets.net/n2o4hgsv6wcx/2AfZNrit52m4WUm4IKUOOE/21323f2171228d475743a62938778c53/ed2cf58359281c503c8fd2d56992b0a9.png?w=800&h=168&q=90',
                        },
                      },
                    },
                  },
                },
                sponsorLevel: {
                  'en-GB': 'Headline',
                },
              },
            },
          ],
        },
      },
      sys: {
        contentType: {
          sys: {
            id: 'sponsorSection',
          },
        },
      },
    },
  },
}
