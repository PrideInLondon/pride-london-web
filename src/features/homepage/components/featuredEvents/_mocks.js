export const mockEvents = [
  {
    node: {
      id: '123abc',
      name: 'Pride Quiz',
      startTime: '2019-06-05T20:00+01:00',
      endTime: '2019-06-05T23:00+01:00',
      eventPriceLow: 3,
      eventsListPicture: {
        file: {
          url: 'test/image.jpg',
          title: 'image alt',
        },
      },
    },
  },
  {
    node: {
      id: '123abcd',
      name: 'Pride Quiz 2',
      startTime: '2019-06-05T20:00+01:00',
      endTime: '2019-06-05T23:00+01:00',
      eventPriceLow: 3,
      eventsListPicture: {
        file: {
          url: 'test/image.jpg',
          title: 'image alt',
        },
      },
    },
  },
]
