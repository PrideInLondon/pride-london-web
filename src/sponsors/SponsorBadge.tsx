import React from 'react'
import { Badge, BadgeImage, NamePlaceholder } from './SponsorBadge.styles'
import { Sponsor } from './PartnersPage.types'

const SponsorBadge = ({
  logo,
  name,
  url,
  level,
  preventLevelSize,
}: Sponsor & { preventLevelSize: boolean }) => (
  <Badge
    level={level}
    href={url}
    rel="noopener noreferrer"
    target="_blank"
    preventLevelSize={preventLevelSize}
  >
    {logo ? (
      <BadgeImage
        src={logo}
        alt={name}
        level={level}
        preventLevelSize={preventLevelSize}
      />
    ) : (
      <NamePlaceholder>{name}</NamePlaceholder>
    )}
  </Badge>
)

export default SponsorBadge
