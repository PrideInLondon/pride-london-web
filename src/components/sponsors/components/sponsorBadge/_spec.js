import React from 'react'
import { shallow, mount } from 'enzyme'
import toJSON from 'enzyme-to-json'
import SponsorBadge from '../sponsorBadge'
import { mockSponsor, mockSponsor2, mockSponsorWithoutLogo } from './_mocks'

describe('<SponsorBadge/>', () => {
  it('renders the component <SponsorBadge />', () => {
    const wrapper = shallow(<SponsorBadge {...mockSponsor} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
    expect(wrapper.prop('href')).toBe(mockSponsor.url)
  })

  it('renders the component <SponsorBadge /> with small size sponsor', () => {
    const wrapper = mount(<SponsorBadge {...mockSponsor2} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders the component <SponsorBadge /> with big size sponsor', () => {
    const wrapper = mount(<SponsorBadge {...mockSponsor} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders the component <SponsorBadge /> without logo', () => {
    const wrapper = mount(<SponsorBadge {...mockSponsorWithoutLogo} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders the component <SponsorBadge /> and prevent level size', () => {
    const wrapper = shallow(<SponsorBadge {...mockSponsor} preventLevelSize />)
    expect(toJSON(wrapper)).toMatchSnapshot()
    expect(wrapper.prop('href')).toBe(mockSponsor.url)
  })
})
