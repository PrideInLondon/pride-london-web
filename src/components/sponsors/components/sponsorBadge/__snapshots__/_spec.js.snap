// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SponsorBadge/> renders the component <SponsorBadge /> 1`] = `
<styles__Badge
  href="https://sponsor1.com"
  level="Headline"
  preventLevelSize={false}
  rel="noopener noreferrer"
  target="_blank"
>
  <styles__BadgeImage
    alt="Sponsor One"
    level="Headline"
    preventLevelSize={false}
    src="/logo1"
  />
</styles__Badge>
`;

exports[`<SponsorBadge/> renders the component <SponsorBadge /> and prevent level size 1`] = `
<styles__Badge
  href="https://sponsor1.com"
  level="Headline"
  preventLevelSize={true}
  rel="noopener noreferrer"
  target="_blank"
>
  <styles__BadgeImage
    alt="Sponsor One"
    level="Headline"
    preventLevelSize={true}
    src="/logo1"
  />
</styles__Badge>
`;

exports[`<SponsorBadge/> renders the component <SponsorBadge /> with big size sponsor 1`] = `
<SponsorBadge
  level="Headline"
  logo="/logo1"
  name="Sponsor One"
  preventLevelSize={false}
  url="https://sponsor1.com"
>
  <styles__Badge
    href="https://sponsor1.com"
    level="Headline"
    preventLevelSize={false}
    rel="noopener noreferrer"
    target="_blank"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styles__Badge-uzqcz1-0",
            "isStatic": false,
            "lastClassName": "czTwOV",
            "rules": Array [
              "border:none;display:inline-flex;align-items:center;justify-content:center;background-color:#f9f9f9;margin:0 10px 10px 0;padding:10px;",
              "@media (min-width:",
              "768",
              "px){",
              "
    margin: 0 30px 30px 0;
  ",
              ";}",
              ";",
              [Function],
              ";",
            ],
          },
          "displayName": "styles__Badge",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styles__Badge-uzqcz1-0",
          "target": "a",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      href="https://sponsor1.com"
      level="Headline"
      preventLevelSize={false}
      rel="noopener noreferrer"
      target="_blank"
    >
      <a
        className="styles__Badge-uzqcz1-0 czTwOV"
        href="https://sponsor1.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <styles__BadgeImage
          alt="Sponsor One"
          level="Headline"
          preventLevelSize={false}
          src="/logo1"
        >
          <StyledComponent
            alt="Sponsor One"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "styles__BadgeImage-uzqcz1-1",
                  "isStatic": false,
                  "lastClassName": "crqdQM",
                  "rules": Array [
                    [Function],
                    ";",
                  ],
                },
                "displayName": "styles__BadgeImage",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "styles__BadgeImage-uzqcz1-1",
                "target": "img",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            level="Headline"
            preventLevelSize={false}
            src="/logo1"
          >
            <img
              alt="Sponsor One"
              className="styles__BadgeImage-uzqcz1-1 crqdQM"
              src="/logo1"
            />
          </StyledComponent>
        </styles__BadgeImage>
      </a>
    </StyledComponent>
  </styles__Badge>
</SponsorBadge>
`;

exports[`<SponsorBadge/> renders the component <SponsorBadge /> with small size sponsor 1`] = `
<SponsorBadge
  level="Silver"
  logo="/logo2"
  name="Sponsor Two"
  preventLevelSize={false}
  url="https://sponsor2.com"
>
  <styles__Badge
    href="https://sponsor2.com"
    level="Silver"
    preventLevelSize={false}
    rel="noopener noreferrer"
    target="_blank"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styles__Badge-uzqcz1-0",
            "isStatic": false,
            "lastClassName": "cZhemG",
            "rules": Array [
              "border:none;display:inline-flex;align-items:center;justify-content:center;background-color:#f9f9f9;margin:0 10px 10px 0;padding:10px;",
              "@media (min-width:",
              "768",
              "px){",
              "
    margin: 0 30px 30px 0;
  ",
              ";}",
              ";",
              [Function],
              ";",
            ],
          },
          "displayName": "styles__Badge",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styles__Badge-uzqcz1-0",
          "target": "a",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      href="https://sponsor2.com"
      level="Silver"
      preventLevelSize={false}
      rel="noopener noreferrer"
      target="_blank"
    >
      <a
        className="styles__Badge-uzqcz1-0 cZhemG"
        href="https://sponsor2.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <styles__BadgeImage
          alt="Sponsor Two"
          level="Silver"
          preventLevelSize={false}
          src="/logo2"
        >
          <StyledComponent
            alt="Sponsor Two"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "styles__BadgeImage-uzqcz1-1",
                  "isStatic": false,
                  "lastClassName": "gsbKBa",
                  "rules": Array [
                    [Function],
                    ";",
                  ],
                },
                "displayName": "styles__BadgeImage",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "styles__BadgeImage-uzqcz1-1",
                "target": "img",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            level="Silver"
            preventLevelSize={false}
            src="/logo2"
          >
            <img
              alt="Sponsor Two"
              className="styles__BadgeImage-uzqcz1-1 gsbKBa"
              src="/logo2"
            />
          </StyledComponent>
        </styles__BadgeImage>
      </a>
    </StyledComponent>
  </styles__Badge>
</SponsorBadge>
`;

exports[`<SponsorBadge/> renders the component <SponsorBadge /> without logo 1`] = `
<SponsorBadge
  level="Silver"
  logo=""
  name="Sponsor Three"
  preventLevelSize={false}
  url="https://sponsor3.com"
>
  <styles__Badge
    href="https://sponsor3.com"
    level="Silver"
    preventLevelSize={false}
    rel="noopener noreferrer"
    target="_blank"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styles__Badge-uzqcz1-0",
            "isStatic": false,
            "lastClassName": "cZhemG",
            "rules": Array [
              "border:none;display:inline-flex;align-items:center;justify-content:center;background-color:#f9f9f9;margin:0 10px 10px 0;padding:10px;",
              "@media (min-width:",
              "768",
              "px){",
              "
    margin: 0 30px 30px 0;
  ",
              ";}",
              ";",
              [Function],
              ";",
            ],
          },
          "displayName": "styles__Badge",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styles__Badge-uzqcz1-0",
          "target": "a",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      href="https://sponsor3.com"
      level="Silver"
      preventLevelSize={false}
      rel="noopener noreferrer"
      target="_blank"
    >
      <a
        className="styles__Badge-uzqcz1-0 cZhemG"
        href="https://sponsor3.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <styles__NamePlaceholder>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "styles__NamePlaceholder-uzqcz1-2",
                  "isStatic": true,
                  "lastClassName": "kxElvH",
                  "rules": Array [
                    "color:#9b9b9b;display:inline-block;margin:0;padding:15px;text-align:center;",
                  ],
                },
                "displayName": "styles__NamePlaceholder",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "styles__NamePlaceholder-uzqcz1-2",
                "target": "h3",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <h3
              className="styles__NamePlaceholder-uzqcz1-2 kxElvH"
            >
              Sponsor Three
            </h3>
          </StyledComponent>
        </styles__NamePlaceholder>
      </a>
    </StyledComponent>
  </styles__Badge>
</SponsorBadge>
`;
